src/java/grl/src/grl/driver/GRL_Driver.java

```diff
@@ -379,84 +412,48 @@ else if (_currentKUKAiiwaState.armControlState().stateType() == grl.flatbuffer.A
 				if (currentMotion != null && _commandInterface!=grl.flatbuffer.KUKAiiwaInterface.FRI) currentMotion.cancel();
 				if(!cancelTeachMode()) continue;
 
+		        // TODO: switch control mode in config change check? or check in individual armControlState?
```

Comment:
I think switching in the config change check is most appropriate. ArmControlState is for the specific commands to be sent at this instant such as the joint angles to move to in the case of MoveArmJointServo.